### ISSUE TITLE
revisit hold-out set configuration

### ISSUE THREAD

**Issue by mastoffel** [alan-turing-institute/autoemulate#161]
* In #155 we added a test set for hyperparameter optimized models to predict.

After talking to @bryanlimy and looking at pycaret, I think we should again restructure to do the following:

* by default, we split into train / test sets
* by default, we return results as average cv scores (on the training set)
* we add a function `predict()` that predicts on the test set, and gives all results as test-set metrics. It can also take additional data and will then predict on that data.
* models will only be trained on training data, but before saving, the user can re-train on the full data using a `finalise_model` or `train_on_full_data` function
